components/hab/src/command/pkg/upload.rs

```diff
@@ -293,3 +269,43 @@ fn attempt_upload_dep(
         ))
     }
 }
+
+fn upload_public_key<T>(
+    ui: &mut UI,
+    token: &str,
+    api_client: &Client,
+    archive: &mut PackageArchive,
+    key_path: T,
+) -> Result<()>
+where
+    T: AsRef<Path>,
+{
+    let hart_header = get_artifact_header(&archive.path)?;
+    let public_keyfile_name = format!("{}.pub", &hart_header.key_name);
+    let public_keyfile = key_path.as_ref().join(&public_keyfile_name);
+
+    let (name, rev) = parse_name_with_rev(&hart_header.key_name)?;
+
+    match api_client.put_origin_key(&name, &rev, &public_keyfile, token, ui.progress()) {
+        Ok(()) => {
+            ui.begin(format!(
+                "Uploading public origin key {}",
+                &public_keyfile_name
+            ))?;
```

Comment:
Previously, this message printed before the call to `api_client.put_origin_key`. Is that an intentional change? Either way seems reasonable to me.

Reply:
yes this was intentional. There might be a cleaner way of handling it, but since we're uploading each dep's key now, that message would get printed N+1 times even if we didn't end up uploading due to the key already existing. Now it only gets printed if we've actually uploaded the key. We could tweak this further but i know @fnichol has some plans for changing this behavior quite a bit more so I figured this is fine for now.